src/input/process/flow.rs

```diff
@@ -16,139 +18,198 @@ const PROCESS_FLOWS_FILE_NAME: &str = "process_flows.csv";
-
-        ensure!(flow.flow != 0.0, "Flow cannot be zero");
-
-        // Check that flow is not infinity, nan, etc.
-        ensure!(
-            flow.flow.is_normal(),
-            "Invalid value for flow ({})",
-            flow.flow
-        );
-
-        // **TODO**: https://github.com/EnergySystemsModellingLab/MUSE_2.0/issues/300
-        ensure!(
-            flow.flow_type == FlowType::Fixed,
-            "Commodity flexible assets are not currently supported"
-        );
+    let mut map: HashMap<ProcessID, ProcessFlowsMap> = HashMap::new();
+    for record in iter {
+        record.validate()?;
+
+        // Get process
+        let id = process_ids.get_id_by_str(&record.process_id)?;
+        let process = processes
+            .get(&id)
+            .with_context(|| format!("Process {id} not found"))?;
+
+        // Get regions
+        let process_regions = process.regions.clone();
+        let record_regions =
+            parse_region_str(&record.regions, &process_regions).with_context(|| {
+                format!("Invalid region for process {id}. Valid regions are {process_regions:?}")
+            })?;
+
+        // Get years
+        let process_year_range = &process.years;
+        let process_years: Vec<u32> = milestone_years
+            .iter()
+            .copied()
+            .filter(|year| process_year_range.contains(year))
+            .collect();
+        let record_years = parse_year_str(&record.year, &process_years).with_context(|| {
+            format!("Invalid year for process {id}. Valid years are {process_years:?}")
+        })?;
 
-        if let Some(flow_cost) = flow.flow_cost {
-            ensure!(
-                (0.0..f64::INFINITY).contains(&flow_cost),
-                "Invalid value for flow cost ({flow_cost}). Must be >=0."
-            )
-        }
+        // Get commodity
+        let commodity = commodities
+            .get(record.commodity_id.as_str())
+            .with_context(|| format!("{} is not a valid commodity ID", &record.commodity_id))?;
 
         // Create ProcessFlow object
-        let process_id = process_ids.get_id_by_str(&flow.process_id)?;
         let process_flow = ProcessFlow {
-            process_id: flow.process_id,
             commodity: Rc::clone(commodity),
-            flow: flow.flow,
-            flow_type: flow.flow_type,
-            flow_cost: flow.flow_cost.unwrap_or(0.0),
-            is_pac: flow.is_pac,
+            flow: record.flow,
+            flow_type: record.flow_type,
+            flow_cost: record.flow_cost.unwrap_or(0.0),
+            is_pac: record.is_pac,
         };
 
-        // Insert into the map
-        flows
-            .entry(process_id)
-            .or_insert_with(Vec::new)
-            .push(process_flow);
+        // Insert flow into the map
+        let entry = map.entry(id.clone()).or_default();
+        for year in record_years {
+            for region in record_regions.clone() {
```

Comment:
You don't need to copy `record_regions` every time:

```suggestion
            for region in record_regions.iter() {
```